incubator/hnc/test/e2e/acting-as.go

```diff
@@ -0,0 +1,42 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "sigs.k8s.io/multi-tenancy/incubator/hnc/pkg/testutils"
+)
+
+var _ = Describe("Acting-as", func() {
+	const (
+		parent = "parent"
+		child  = "child"
+	)
+
+	BeforeEach(func() {
+		CleanupNamespaces(parent, child)
+	})
+
+	AfterEach(func() {
+		CleanupNamespaces(parent, child)
+	})
+
+	It("should allow acting as different service accounts", func() {
+		// create the namespaces
+		MustRun("kubectl create ns", parent)
+		MustRun("kubectl create ns", child)
+
+		// fail to set parent
+		MustNotRun("kubectl hns set", child, "--parent", parent, "--as system:serviceaccount:"+child+":default")
+
+		// set the proper roles giving the required permissions
+		MustRun("kubectl -n", child, "create role "+child+"-hnc-role --verb=get,create,update --resource=HierarchyConfiguration")
```

Comment:
I think you can use the built-in `admin` clusterrole instead of creating a new role here; this will simplify the test. You can grant cluster roles in per-namespace role bindings, which only grants the permissions within a single namespace.